perceval/components/abstract_component.py

```diff
@@ -113,16 +114,17 @@ def get_parameters(self, all_params: bool = False, expressions = False) -> list[
             If False, returns the raw parameters that make up the expressions only. Default `False`.
         """
         param_list = []
-        for param in self._params.values():
+        for param in self._raw_params.values():
             if all_params or not param.fixed:
                 if isinstance(param, Expression):
                     if expressions:
+                        # Add raw expression
                         if param not in param_list:
-                            param_list.append(param)
+                            param_list.append(param) 
                     else:
-                        for p in param.parameters:
-                            if p not in param_list:
-                                param_list.append(p)
+                        # Add subparameters within expression
+                        param_list = list(set(param_list) | param._params)
```

Comment:
Beware that using `set` can lead to random ordering of the results (though I agree that this is faster), whereas this method currently gives the parameters in the order they were added.